pandas/tseries/base.py

```diff
@@ -67,7 +67,7 @@ def __contains__(self, key):
         try:
             res = self.get_loc(key)
             return np.isscalar(res) or type(res) == slice or np.any(res)
-        except (KeyError, TypeError):
+        except (KeyError, TypeError, ValueError):
```

Comment:
Honestly, I suspect doing that will break more stuff :). Probably not worth it unless you want to dive deep into the rabbit hole to clean things up...